src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -180,13 +181,16 @@ UCS_CLASS_INIT_FUNC(uct_rdmacm_cm_ep_t, const uct_ep_params_t *params)
 
     if (params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR) {
         status = uct_rdamcm_cm_ep_client_init(self, params);
-    } else {
-        ucs_assert(params->field_mask & UCT_EP_PARAM_FIELD_CONN_REQUEST);
+    } else if (params->field_mask & UCT_EP_PARAM_FIELD_CONN_REQUEST) {
         status = uct_rdamcm_cm_ep_server_init(self, params);
+    } else {
+        ucs_error("either UCT_EP_PARAM_FIELD_SOCKADDR or UCT_EP_PARAM_FIELD_CONN_REQUEST "
+                  "have to be provided in the ep params (uct_ep_params_t)");
```

Comment:
"either UCT_EP_PARAM_FIELD_SOCKADDR or UCT_EP_PARAM_FIELD_CONN_REQUEST has to be provided"